### DISCUSSION TITLE
Adding custom kernel to Navier-Stokes module

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#23270] [Q&A Modules: Navier-Stokes] [answered]
Hello,

I developed a custom kernel myself. Is there any way that during compilation this can be added to navier_stokes-opt, so it can be compiled with other kernels? For example, I added the .h and .c file of my custom kernel to Navier-stokes module and it did not work (include/kernels and src/kernels folders). What is the correct means of adding our custom kernels to existing MOOSE kernels and compile everything together?

Thanks,
Ali

**Comment by GiudGiud** [comment, marked as answer]
https://github.com/idaholab/moose/discussions/23254#discussioncomment-4802183
